src/app/_services/depublication.service.ts

```diff
@@ -0,0 +1,116 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { apiSettings } from '../../environments/apisettings';
+import { MoreResults, RecordPublicationInfo, Results, SortParameter } from '../_models';
+
+import { collectResultsUptoPage } from './service-utils';
+import { ErrorService } from './error.service';
+
+@Injectable({ providedIn: 'root' })
+export class DepublicationService {
+  constructor(private readonly http: HttpClient, private readonly errors: ErrorService) {}
+
+  /** handleUploadEvents
+  /*  record upload events
+  /*  @param {HttpEvent} event - event from the server
+  */
+  handleUploadEvents(event: HttpEvent<Object>): boolean {
+    if (event.type === HttpEventType.Response) {
+      return true;
+    } else {
+      if (event.type === HttpEventType.UploadProgress) {
+        console.log(`File upload: ${event.loaded} / ${event.total}`);
+      }
+      return false;
+    }
+  }
+
+  /** setPublicationFile
+  /*  post publication information
+  /*  @param {string} datasetId - the dataset id
+  /*  @param {File} file - file of record urls
+  */
+  setPublicationFile(datasetId: string, file: File): Observable<boolean> {
+    const url = `${apiSettings.apiHostCore}/depublished_records/${datasetId}`;
+    const formData = new FormData();
+
+    formData.append('depublicationFile', file);
+
+    return this.http
+      .post(url, formData, {
+        headers: {
+          'Content-Type': 'multipart/form-data'
+        },
+        observe: 'events',
+        params: {
+          clientFilename: file.name,
+          mimeType: file.type
+        },
+        reportProgress: true
+      })
+      .pipe(map(this.handleUploadEvents))
+      .pipe(this.errors.handleRetry());
+  }
+
+  /** setPublicationInfo
+  /*  post publication information
+  /*  @param {string} datasetId - the dataset id
+  /*  @param {string} toDepublish - depublication record urls
+  */
+  setPublicationInfo(datasetId: string, toDepublish: string): Observable<boolean> {
+    const url = `${apiSettings.apiHostCore}/depublished_records/${datasetId}`;
+    return this.http
+      .post<boolean>(
+        url,
+        { toDepublish },
+        {
+          headers: {
+            'Content-Type': 'text/plain'
+          }
+        }
+      )
+      .pipe(this.errors.handleRetry());
+  }
+
+  /** parseSortParameter
+  /*  return string representation of a SortParameter or an empty string
+  /*  @param {SortParameter} p - optional
+  */
+  parseSortParameter(p?: SortParameter): string {
+    return p ? `&sortField=${p.field}&sortDirection=${p.direction.toLowerCase()}` : '';
+  }
+
+  /** getPublicationInfo
+  /*  retrieve publication information
+  /*  @param {string} datasetId - the dataset id
+  /*  @param {number} page - the pagination target
+  /*  @param {SortParameter} sort - optional - sort parameter
+  */
+  getPublicationInfo(
```

Comment:
I added the option of searching as well to this service. The query parameter is called `searchQuery`.